x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -211,41 +198,13 @@ public AuthenticationToken token(final ThreadContext threadContext) {
         final SignedJWT signedJWT;
         try {
             signedJWT = SignedJWT.parse(userCredentials.toString());
+            // trigger claim set parsing
+            signedJWT.getJWTClaimsSet();
```

Comment:
If we use the raw middle bits of the claim set as suggested in the other comment this probably is not needed.